src/app/games/epaminondas/epaminondas.component.ts

```diff
@@ -125,276 +129,133 @@ export class EpaminondasComponent extends RectangularGameComponent<EpaminondasRu
-            phalanxDirection = phalanxLanding;
-        }
-        if (phalanxDirection !== phalanxLanding) {
-            return this.cancelMove(EpaminondasFailure.SQUARE_NOT_ALIGNED_WITH_PHALANX());
-        }
-        if (this.board[y][x] === player) {
-            return this.thirdClickOnPlayerPiece(clicked, phalanxDirection, player);
-        } else {
-            const phalanxSize: number = this.firstPiece.get().getDistance(this.lastPiece.get()) + 1;
-            const stepSize: number = this.lastPiece.get().getDistance(clicked);
-            const move: EpaminondasMove = new EpaminondasMove(this.firstPiece.get().x,
-                                                              this.firstPiece.get().y,
-                                                              phalanxSize,
-                                                              stepSize,
-                                                              phalanxDirection);
-            return this.chooseMove(move);
-        }
+        return this.chooseMoveFromClicks(clicked);
     }
-    private async thirdClickOnPlayerPiece(clicked: Coord,
-                                          phalanxDirection: Direction,
-                                          player: Player)
-    : Promise<MGPValidation>
-    {
-        this.lastPiece = MGPOptional.of(clicked);
-        const phalanxSize: number = this.firstPiece.get().getDistance(clicked) + 1;
-        const incompleteMove: EpaminondasMove = new EpaminondasMove(this.firstPiece.get().x,
-                                                                    this.firstPiece.get().y,
-                                                                    phalanxSize,
-                                                                    1,
-                                                                    phalanxDirection);
-        const phalanxValidity: MGPValidation = EpaminondasRules.getPhalanxValidity(this.node.gameState,
-                                                                                   incompleteMove);
-        if (phalanxValidity.isFailure()) {
-            return this.cancelMove(phalanxValidity.getReason());
-        } else {
-            this.phalanxMiddles = this.firstPiece.get().getCoordsToward(this.lastPiece.get());
-            this.validExtensions = this.getValidExtensions(player);
-            this.phalanxValidLandings = this.getPhalanxValidLandings();
-            return MGPValidation.SUCCESS;
+
+    private chooseMoveFromClicks(clicked: Coord): Promise<MGPValidation> {
+        const direction: Direction = this.firstPiece.get().getDirectionToward(clicked).get();
+        const phalanxSize: number = this.countPhalanxSize(direction);
+        const stepSize: number = this.getStepSize(clicked, phalanxSize);
+        if (stepSize > phalanxSize) {
+            return this.cancelMove(EpaminondasFailure.PHALANX_CANNOT_JUMP_FURTHER_THAN_ITS_SIZE(stepSize, phalanxSize));
```

Comment:
c'est un message montrable àl'utilisateur, il faut donc le formuler en humain (là il commence par "Cannot", peut-être faire "You tried to XXX, but should YYY")